File: price-comparison-client1/src/bll/service/model/IProductService.ts
import type { IProduct } from '@/dal/domain/IProduct'
import type { IServiceResult } from '@/bll/service/model/IServiceResult'
import type { IProductImport } from '@/dal/domain/IProductImport'

/**
 * @author Ahto Jalak
 * @since 12.04.2023
 */
export interface IProductService {
    findByName (searchName: string): Promise<IServiceResult<IProduct[]>>

    add (productInfo: IProduct): Promise<IServiceResult<IProduct>>

    getAll (): Promise<IServiceResult<IProduct[]>>

    import (entity: IProductImport): Promise<IServiceResult<IProduct[]>>

    getById (productId: number): Promise<IServiceResult<IProduct>>
}

File: public-api/src/main/java/com/shopleech/publicapi/bll/service/model/IProductService.java
package com.shopleech.publicapi.bll.service.model;

import com.shopleech.publicapi.domain.Product;
import com.shopleech.publicapi.dto.v1.ProductImportDTO;
import com.shopleech.publicapi.dto.v1.ProductImportItemDTO;

import java.util.List;

/**
 * @author Ahto Jalak
 * @since 04.03.2023
 */
public interface IProductService {
    Product get(Integer id) throws Exception;

    List<Product> getAll();

    List<Product> getAllByKeyword(String keyword);

    String importProducts(String token, ProductImportDTO productImportItems);

    Product add(Product product);

    Product update(Integer id, Product product);

    Integer remove(Integer id);

    List<Product> getAllByProduct(Product product);
}
